File: model/pvap_ch4.f
      double precision function pvap_ch4( t )
c
c     calculate saturation vapor pressure (dyne/cm^2) over CH4
c
c     input temperature in K
c
c     Rages Jul-2000 
c
      implicit none

      integer ic
      double precision a(2),b(2),c(2)

c     quantities to calculate methane vapor pressure     
c     (WHAT ARE THE UNITS?)
c
c     AMR   -- molecular weight / ideal gas constant
c     TCRIT -- triple point temperature
c     PCRIT --    "     "   pressure
c     AS    -- specific heat at constant pressure ( gas - solid )
c     AL    --    "      "   "     "        "     ( gas - liquid )
c     ALS   -- latent heat of sublimation
c     ALV   --   "     "   "  vaporization
c
      double precision t,amr,tcrit,pcrit,as,al,als,alv
      parameter( AMR = 16.043 / 8.3143, TCRIT = 90.68, PCRIT = .11719 )
      parameter( AS = 2.213 - 2.650, AL = 2.213 - 3.370 )
      parameter( ALS = 611.10, ALV = 552.36 )
c
c     ic=1: temperature below triple point
c     ic=2: temperature above triple point
c
      ic = 1
      if (t.gt.tcrit) ic = 2

      C(1) = - AMR * AS
      C(2) = - AMR * AL
      B(1) = - AMR * ( ALS + AS * TCRIT )
      B(2) = - AMR * ( ALV + AL * TCRIT )
      A(1) = PCRIT * TCRIT ** ( -C(1) ) * EXP( -B(1) / TCRIT )
      A(2) = PCRIT * TCRIT ** ( -C(2) ) * EXP( -B(2) / TCRIT )

      pvap_ch4 = A(IC) * t**C(IC) * EXP( B(IC) / t )
   
      pvap_ch4= pvap_ch4*1e6    ! convert from bars to dyne/cm^2
c     pvap_ch4= pvap_ch4*1e1    ! convert from Pa to dyne/cm^2: T ~ 240 K
c     pvap_ch4= pvap_ch4*1e3    ! convert from mb to dyne/cm^2
c     pvap_ch4= pvap_ch4*1e2    ! nonsense, but gives cloud base T ~ 140 K

      return
      end

File: python_version/pvap_ch4.py
import numpy as np

class VaporPressure:
    @staticmethod
    def pvap_ch4(t):
        """
        Calculate saturation vapor pressure (dyne/cm^2) over CH4.
        Input: t (temperature in K)
        Output: vapor pressure in dyne/cm^2
        """
        AMR = 16.043 / 8.3143
        TCRIT = 90.68
        PCRIT = 0.11719
        AS = 2.213 - 2.650
        AL = 2.213 - 3.370
        ALS = 611.10
        ALV = 552.36

        # ic=1: temperature below triple point, ic=2: above
        ic = 1 if t <= TCRIT else 2

        C = np.zeros(2)
        B = np.zeros(2)
        A = np.zeros(2)

        C[0] = -AMR * AS
        C[1] = -AMR * AL
        B[0] = -AMR * (ALS + AS * TCRIT)
        B[1] = -AMR * (ALV + AL * TCRIT)
        A[0] = PCRIT * TCRIT ** (-C[0]) * np.exp(-B[0] / TCRIT)
        A[1] = PCRIT * TCRIT ** (-C[1]) * np.exp(-B[1] / TCRIT)

        idx = ic - 1  # Python uses 0-based indexing
        pvap_ch4 = A[idx] * t ** C[idx] * np.exp(B[idx] / t)
        pvap_ch4 = pvap_ch4 * 1e6  # convert from bars to dyne/cm^2

        return pvap_ch4

# Example usage:
# vp = VaporPressure.pvap_ch4(100)
# print(vp)